core/src/main/java/tc/oc/pgm/action/actions/WeatherAction.java

```diff
@@ -0,0 +1,40 @@
+package tc.oc.pgm.action.actions;
+
+import tc.oc.pgm.action.WeatherType;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.modules.WeatherMatchModule;
+
+public class WeatherAction extends AbstractAction<Match> {
+
+  private final Enum state;
+
+  public WeatherAction(Enum state) {
+    super(Match.class);
+    this.state = state;
+  }
+
+  public void trigger(Match match) {
+    WeatherMatchModule weatherMatchModule = match.getModule(WeatherMatchModule.class);
+
+    weatherMatchModule.allowChange();
+
+    switch (state) {
+      case WeatherType.CLEAR:
+        match.getWorld().setStorm(false);
+        break;
+      case WeatherType.RAIN:
+        match.getWorld().setStorm(true);
+        match.getWorld().setThundering(false);
+        match.getWorld().setThunderDuration(0);
+        break;
+      case WeatherType.THUNDER:
+        match.getWorld().setStorm(true);
+        match.getWorld().setThundering(true);
+        break;
+      default:
+        break;
```

Comment:
consider expanding the WeatherType enum to contain the storm and thundering true/false attributes, that way you can also tuck away all this logic into there where you can just call `WeatherType.RAIN.enable()` and it avoids the whole switch being needed